generic/zookeepertcl.c

```diff
@@ -1634,25 +1585,33 @@ zootcl_children_subcommand(Tcl_Interp *interp, int objc, Tcl_Obj *CONST objv[],
 
 
 	if (callbackObj == NULL) {
-		// synchronous request
-		// invoke the special async handler for sync requests
-		// so the event loop will still be alive
-		zootcl_syncCallbackContext *zsc = (zootcl_syncCallbackContext *)ckalloc (sizeof (zootcl_syncCallbackContext));
-		zsc->zo = zo;
-		zsc->syncDone = 0;
-		status = zoo_awget_children (zh, path, wfn, watcherCallbackObj, zootcl_sync_strings_completion_callback, zsc);
-		if (zootcl_wait (zo, zsc) == TCL_ERROR) {
-			ckfree (zsc);
-			return TCL_ERROR;
-		}
+		struct String_vector *strings = (struct String_vector *)ckalloc (sizeof (struct String_vector));
+		status = zoo_wget_children(zh, path, wfn, (void *)watcherCallbackObj, strings);	
 
-		if (status != ZOK) {
-			ckfree (zsc);
+		if (status != ZOK && status != ZNONODE) {
+			ckfree (strings);
 			return zootcl_set_tcl_return_code (interp, status);
+		} else if (status == ZNONODE) {
+			// do not consider a non-existent path to be an error in this case
+			status = ZOK;
+			strings->count = 0;
 		}
-
-		Tcl_SetObjResult (interp, zsc->dataObj);
-		ckfree (zsc);
+		
+		int count = strings ? strings->count : 0;
```

Comment:
I don't think strings can be null here.

Reply:
Good catch: I'm gotten rid of the use of `count`.